### PR TITLE
[1137] Adding a baseline karma file

### PR DESCRIPTION
**Pull request by Mutmatt**
In a lot of the `karma.conf` files there was a reference to `const getBaseKarmaConfig = require('../../karma.conf');` That file did not exist so adding it to the root of the project allows successful runs of the rest of the test suite.

Issue ref #1137

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['Chrome'],
+
+
+    // Continuous Integration mode
+    // if true, Karma captures browsers, runs the tests and exits
+    singleRun: false,
```
Comment (NothingEverHappens) [inline karma.conf.js]: This would probably break CI?
AFAIK ng passes this setting already, so it's safe to drop

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    colors: true,
+
+
+    // level of logging
+    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
+    logLevel: config.LOG_INFO,
+
+
+    // enable / disable watching file and executing tests whenever any file changes
+    autoWatch: false,
```
Comment (NothingEverHappens) [inline karma.conf.js]: This prob can also be dropped

File: karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    basePath: '',
+
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['jasmine'],
+
+
+    // list of files / patterns to load in the browser
+    files: [
```
Comment (NothingEverHappens) [inline karma.conf.js]: We probably should be safe to drop those defaults and make this file a bit cleaner and shorted?

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+
+    // test results reporter to use
+    // possible values: 'dots', 'progress'
+    // available reporters: https://npmjs.org/browse/keyword/karma-reporter
+    reporters: ['progress'],
+
+
+    // web server port
+    port: 9876,
```
Comment (NothingEverHappens) [inline karma.conf.js]: This seems like a default value, can this be dropped?

**Comment by NothingEverHappens** [comment]
Awesome, thanks a lot for adding it, it would be cool to have the very minimum required configuration in this file?

File: karma.conf.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Karma configuration
+// Generated on Fri Oct 18 2019 15:00:24 GMT-0500 (Central Daylight Time)
```
Comment (NothingEverHappens) [inline karma.conf.js]: Prob not that value in this comment as well

Comment (Mutmatt) [reply karma.conf.js]: Lol on it

**Comment by NothingEverHappens** [comment]
I have a feeling that this file wouldn't be working really well as angular probably requires some specific config which would be missing from the resulting double config. 

On the other side this PR is for sure a step forward, so let's merge it

**Comment by Mutmatt** [comment]
This is causing some issues running local tests. 

`ng "test" "kirjs" "--watch=true"
18 10 2019 17:25:01.015:ERROR [config]: Error in config file!
 TypeError: Cannot read property 'set' of undefined
    at module.exports (/Users/mutmatt/Source/codelab/karma.conf.js:2:10)
    at module.exports (/Users/mutmatt/Source/codelab/apps/kirjs/karma.conf.js:8:22)
    at Object.parseConfig (/Users/mutmatt/Source/codelab/node_modules/karma/lib/config.js:393:5)
    at new Server (/Users/mutmatt/Source/codelab/node_modules/karma/lib/server.js:66:24)
    at Observable.rxjs_1.Observable.subscriber [as _subscribe] (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/src/karma/index.js:100:29)
    at Observable._trySubscribe (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/Observable.ts:231:19)
    at Observable.subscribe (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/Observable.ts:212:14)
    at /Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/util/subscribeTo.ts:22:23
    at Object.subscribeToResult (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/util/subscribeToResult.ts:25:29)
    at SwitchMapSubscriber._innerSub (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/operators/switchMap.ts:122:30)
    at SwitchMapSubscriber._next (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/operators/switchMap.ts:111:10)
    at SwitchMapSubscriber.Subscriber.next (/Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/Subscriber.ts:99:12)
    at /Users/mutmatt/Source/codelab/node_modules/@angular-devkit/build-angular/node_modules/rxjs/src/internal/util/subscribeToPromise.ts:8:20
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)`

need to fix some stuff up first before this goes through

**Comment by NothingEverHappens** [comment]
Cool, I think the best way to fix would be to get some inspiration from working karma configs, e.g. the one in apps/codelab (or drop karma and just switch to jest)

**Comment by kirjs** [comment]
Given #1176, is this PR needed?

**Comment by Mutmatt** [comment]
> Given #1176, is this PR needed?

Nope, 🔪 the pr :)

**Pull request closed without merging**